packages/cli/test/integration/controllers/oauth/oauth2.api.test.ts

```diff
@@ -0,0 +1,107 @@
+import { Container } from 'typedi';
+import { response as Response } from 'express';
+import nock from 'nock';
+import { parse as parseQs } from 'querystring';
+
+import type { CredentialsEntity } from '@db/entities/CredentialsEntity';
+import type { User } from '@db/entities/User';
+import { CredentialsHelper } from '@/CredentialsHelper';
+import { OAuth2CredentialController } from '@/controllers/oauth/oAuth2Credential.controller';
+
+import { createOwner } from '@test-integration/db/users';
+import { saveCredential } from '@test-integration/db/credentials';
+import * as testDb from '@test-integration/testDb';
+import { setupTestServer } from '@test-integration/utils';
+import type { SuperAgentTest } from '@test-integration/types';
+
+describe('OAuth2 API', () => {
```

Comment:
Do we need tests for Oauth1 as well? Or is the support for that deprecated?

Reply:
TBH, everytime I've asked about oauth1, I've only heard that some folks might be using it for Twitter, but I haven't actually seen anyone use it. 
In the past we've had bugs in oauth1 flow, and no one reported them either.
In the last few months, we've refactored this code to make oauth1 flow use a lot more code shared with oauth2 flow. 
So technically this already covers quite a bit of the oauth1 flow as well.
But, I don't think it's worth the effort to write dedicated integration tests for oauth1.
We can add these tests when someone reports a bug in that code.